### PR TITLE
LPS-173162 Cache extracted text when indexing

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/browse/LPS-173162

Reindexing documents can be inefficient for large documents, as extracting text can be a time consuming process in some cases (e.g. large PDF files).

Cache the extracted text in the store and reuse if possible.  These cached files will be delete on update/deletion of a file version.

# How do I test it?

There should be no user visible changes, so there's really nothing to test.  If all automated tests pass this should be ready to go.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by austinchiang** [comment]
There are some failing DL integration tests which look relevant https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2166087802

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -2488,6 +2488,15 @@ private void _copyFileEntryMetadata(
 		}
 	}
 
+	private void _deleteFile(
+			long companyId, long repositoryId, String name, String versionLabel)
+		throws PortalException {
+
+		DLStoreUtil.deleteFile(companyId, repositoryId, name, versionLabel);
+		DLStoreUtil.deleteFile(
```
Comment (AliciaGarciaGarcia) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2496]: I'm not sure if this file exist in all the cases. so previously maybe we should check if `versionLabel+".index"` exists with a `DLStoreUtil.hasFile` ?

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java (hunk 1 above)
Comment (adolfopa) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2496]: It is not necessary. The `deleteFile` methods will do nothing if the files don't exist.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
Test failures were legit. Pushed a fix.

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/index/contributor/DLFileEntryModelDocumentContributor.java [hunk 2]
```diff
@@ -255,6 +250,45 @@ private String _extractDDMContent(
...
+					dlFileEntry.getCompanyId(),
+					dlFileEntry.getDataRepositoryId(), dlFileEntry.getName(),
+					_getIndexVersionLabel(dlFileEntry)));
+		}
+
+		InputStream inputStream = _getInputStream(dlFileEntry);
+
+		if (inputStream == null) {
+			return null;
+		}
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/index/contributor/DLFileEntryModelDocumentContributor.java:271]: Is it possible a scenario where we have a null `inputStream` but we do have the file in the DLStore? In that case, would it be OK to return what we have in the DLStore?

Comment (adolfopa) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/index/contributor/DLFileEntryModelDocumentContributor.java:271]: If there's a file, the inputStream will never be null. If the file is empty, then the resulting text will be the empty string (which is perfectly fine).

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/index/contributor/DLFileEntryModelDocumentContributor.java (hunk 2 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/index/contributor/DLFileEntryModelDocumentContributor.java:271]: 👍

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1277401_6599

**Comment by adolfopa** [comment]
ci:test:relevant

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:stop

**Comment by adolfopa** [comment]
Squashed SF fix (only build.gradle in https://github.com/liferay-lima/liferay-portal/pull/3927/commits/23e94c56883818b7e255cd3286e50e0db287ca0e affected).

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1317401_6784

**Comment by marcogalluzzi** [comment]
ci:reevaluate:1317401_6784

**Comment by marcogalluzzi** [comment]
Hi @austinchiang can you please check if there is something we can do to solve the test errors? Thanks

**Comment by austinchiang** [comment]
ci:test:sf

**Comment by austinchiang** [comment]
Sent to BChan https://github.com/brianchandotcom/liferay-portal/pull/129284

**Pull request closed without merging**